dashboard/components/DdlGraph.tsx

```diff
@@ -0,0 +1,269 @@
+    const includedDdlIds = new Set<string>()
+    for (const ddlBox of ddlDependencyDag) {
+      let ddlId = ddlBox.id
+      layoutDdlResult.set(ddlId, {
+        ddl_name: ddlBox.ddl_name,
+        schema_name: ddlBox.schema_name,
+      })
+      includedDdlIds.add(ddlId)
+    }
+
+    const ddlLayout = layoutItem(
+      ddlDependencyDag.map(({ width: _1, height: _2, id, ...data }) => {
+        return { width: ddlMarginX, height: ddlMarginY, id, ...data }
+      }),
+      ddlDistanceX,
+      ddlDistanceY
+    )
+
+    let svgWidth = 0
+    let svgHeight = 0
+    ddlLayout.forEach(({ x, y }) => {
+      svgHeight = Math.max(svgHeight, y + ddlMarginX)
+      svgWidth = Math.max(svgWidth, x + ddlMarginY)
+    })
+    const edges = generateDdlEdges(ddlLayout)
+
+    return {
+      layoutResult: ddlLayout,
+      svgWidth,
+      svgHeight,
+      edges,
+    }
+  }, [ddlDependency])
+
+  const {
+    svgWidth,
+    svgHeight,
+    edges: ddlEdgeLayout,
+    layoutResult: ddlLayout,
+  } = ddlDependencyDagCallback()
+
+  useEffect(() => {
+    if (ddlLayout) {
+      const svgNode = svgRef.current
+      const svgSelection = d3.select(svgNode)
+
+      // Ddls
+      const applyDdl = (gSel: DdlSelection) => {
+        gSel.attr("transform", ({ x, y }) => `translate(${x}, ${y})`)
+
+        // Render ddl name
+        {
+          let text = gSel.select<SVGTextElement>(".text-frag-id")
+          if (text.empty()) {
+            text = gSel.append("text").attr("class", "text-frag-id")
+          }
+
+          text
+            .attr("fill", "black")
+            .text(({ ddl_name, schema_name }) => `${schema_name}.${ddl_name}`)
+            .attr("font-family", "inherit")
+            .attr("text-anchor", "middle")
+            .attr("dx", ddlNameMarginX)
+            .attr("dy", ddlNameMarginY)
+            .attr("fill", "black")
+            .attr("font-size", 12)
+        }
+
+        // Render ddl node
+        {
+          let circle = gSel.select<SVGCircleElement>("circle")
+          if (circle.empty()) {
+            circle = gSel.append("circle")
+          }
+
+          circle.attr("r", 20).attr("fill", (_) => {
+            const weight = "500"
+            return theme.colors.gray[weight]
+          })
+
+          circle
+            .attr("cx", ddlMarginX / 2)
+            .attr("cy", ddlMarginY / 2)
+            .attr("fill", "white")
+            .attr("stroke-width", 1)
+            .attr("stroke", theme.colors.gray[500])
+        }
+      }
+
+      const createDdl = (sel: Enter<DdlSelection>) =>
+        sel.append("g").attr("class", "ddl").call(applyDdl)
+
+      const ddlSelection = svgSelection
+        .select<SVGGElement>(".ddls")
+        .selectAll<SVGGElement, null>(".ddl")
+        .data(ddlLayout)
+      type DdlSelection = typeof ddlSelection
+
+      ddlSelection.enter().call(createDdl)
+      // TODO(kwannoel): Is this even needed? I commented it out.
+      // ddlSelection.call(applyDdl)
```

Comment:
This is copied over from `FragmentGraph.tsx`. But I saw that `createDdl` already invokes `applyDdl`. So not sure if actually needed.

Reply:
Nvm, seems needed, otherwise the graph weights not rendered.